lib/bolt/pal/yaml_plan.rb

```diff
@@ -91,7 +93,9 @@ class CodeLiteral < EvaluableString
         def evaluate(scope, evaluator)
           parse_result = evaluator.parse_string(@value)
 
-          evaluator.evaluate(scope, parse_result)
+          scope.with_local_scope({}) do
+            evaluator.evaluate(scope, parse_result)
```

Comment:
I feel like parse_result could be a method and evaluate could live in the base class.